File: webtttn (1)/webtttn/src/main/java/com/webtttn/exercise03/service/CategoryService.java
package com.webtttn.exercise03.service;

import java.util.List;
import java.util.UUID;

import org.springframework.web.multipart.MultipartFile;

import com.webtttn.exercise03.entity.Category;
import com.webtttn.exercise03.entity.Gallery;

public interface CategoryService {

    Category addCategory(Category category);
    
    Category saveImage(UUID productId, MultipartFile file,int i);
    
    List<Category> saveImages(UUID productId, MultipartFile[] files);

    Category getCategoryById(UUID categoryId);

    List<Category> getAllCategories();

    Category updateCategory(UUID categoryId, Category category);

    void deleteCategory(UUID categoryId);

    List<Category> getCategoriesByParentId(UUID parentId);

    List<Category> getRootCategories();

    void deleteImageFile(String imagePath);
}

File: frontend/src/services/CategoryService.js

import { callApi } from "../httpAxios";

function GET_ALL_CATEGORIES_WITHOUT_PARENT(endpoint) {
    return callApi(endpoint, 'GET');
}


function GET_CATEGORIES_BY_PARENT_ID(endpoint, parentId) {
    return callApi(`${endpoint}/${parentId}`, 'GET');
}

function GET_ALL_CATEGORIES(endpoint) {
    return callApi(endpoint, "GET");
}

const categoryservice = {
    GET_ALL_CATEGORIES_WITHOUT_PARENT: GET_ALL_CATEGORIES_WITHOUT_PARENT,
    GET_CATEGORIES_BY_PARENT_ID: GET_CATEGORIES_BY_PARENT_ID,
    GET_ALL_CATEGORIES: GET_ALL_CATEGORIES
}
export default categoryservice;

